### PR TITLE
database migration guide

### PR DESCRIPTION
**Pull request by melissa-hale**
add migration guide for plugins to database services

### CODE REVIEW

File: src/docs/guides/database-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: Migrating from Legacy Plugins to Database Services
+---
+
+We've introduced a more secure and flexible way to deploy databases on Railway.  The purpose of this guide is to provide an overview of why we are making these changes, what they mean for you, and how you can migrate.
+
+## What is Changing?
+### Legacy Plugins
+In the early stages of Railway, we introduced `plugins`, a foundational part of our offering. They were databases deployed from Docker images with a fixed version, no data persistence, and no way to customize the database to your needs.
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:9]: Big nit. But the code formatting of plugins looks a bit strange for non-code things
```suggestion
In the early stages of Railway, we introduced "Plugins", a foundational part of our offering. They were databases deployed from Docker images with a fixed version, no data persistence, and no way to customize the database to your needs.
```

File: src/docs/guides/database-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Database Services
+Some of our newest features - Volumes, Docker Image Deployments, TCP Proxy - open the door for deploying databases as services, complete with data persistence, flexible configuration and management, and connectivity from the external network.
+
+## Why Migrate?
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:14]: Can maybe also include
- Updated versions of database images
- Ability to use the private network

File: src/docs/guides/database-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Migration Process
+
+Here is what will happen when you choose to migrate via our one-click migration process - 
+
+  *screenshot of button?*
+
+1. **One-click button** - When you access the plugin settings from within your project canvas, you will see a prompt to migrate.
+2. **Modal** - Once you click on the button, a modal will appear, detailing the steps that will be taken on your behalf to migrate your data.
+3. **Services Stop** - Any service within your project that is connected via a variable reference to the plugin will be stopped for the duration of the migration to prevent data loss or corruption.
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:30]: Should we include that if you are using the database url externally that you should also stop writing to the database

File: src/docs/guides/database-migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+4. **Database Service** - A new database service with an attached volume will be deployed.
+5. **Data Transfer** - A separate service will be deployed that will execute a migration script.  The script will connect to your old plugin, create a dump of the data, and then transfer it to the new database service.
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:32]: Should mention that these scripts are open-source and link to their repos

File: src/docs/guides/database-migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+6. **Reference Update** - All variable references within other services that point to the old plugin will be updated to the new database service.
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:33]: And we will also redeploy those services

File: src/docs/guides/database-migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+_**NOTE: Your old plugin will NOT be deleted automatically.  You should perform the necessary actions to ensure data consistency between the plugin and the new database service.**_
```
Comment (coffee-cup) [inline src/docs/guides/database-migration-guide.md:35]: Should also mention that the user needs to delete the plugin when the migration is completed

File: src/docs/guides/database-migration-guide.md (hunk 4 above)
Comment (melissa-hale) [reply src/docs/guides/database-migration-guide.md]: Just need the links when you have them but added placeholders

File: src/docs/guides/database-migration-guide.md (hunk 2 above)
Comment (melissa-hale) [reply src/docs/guides/database-migration-guide.md:16]: let me know what you think of the update @coffee-cup and whether that's what you had in mind

**Pull request merged**